src/sentry/models/debugfile.py

```diff
@@ -363,6 +363,23 @@ def _analyze_progard_filename(filename: str) -> Optional[str]:
         return None
 
 
+@region_silo_only_model
+class ProguardArtifactRelease(Model):  # type: ignore
+    __include_in_export__ = False
+
+    organization_id = BoundedBigIntegerField(db_index=True)
+    project_id = BoundedBigIntegerField(db_index=True)
+    release_name = models.CharField(max_length=250)
+    proguard_uuid = models.UUIDField()
```

Comment:
As this is "just a uuid", it does not have a link back to the uploaded proguard file.
We are working on eventually expiring and auto-removing expiring debug files, which also applies to proguard files.

So the underlying proguard `projectdsymfile` may be going away, and you will be left with some "dead" rows here.

Reply:
This detail highly depends on the reading assumptions that the application layer does. Seeing how information is queried, what @Swatinem says it's very good. Either we enforce constraints at the db level or we make sure that the application will do it.

I personally would prefer to link to the file that contains the `proguard_uuid` and have the db handle the consistency requirements. Of course you would also keep the `proguard_uuid`, similarly to how the `DebugIdArtifactBundle` table is designed.

Reply:
I would definitely link to the `ProjectDebugFile`, and also query it on `post`. Right now you can just `post` any random uuid / release_name, and there is no validation whatsoever, and no cleanup. So malicious users could just spam random uuids to fill up the database with junk.

Reply:
Yup, makes sense, thx for taking a look

Reply:
The problem still exists, but to a lesser extent with loose release association, also relevant for artifact bundles (@iambriccardo):
malicious users could spam random release names and fill up the database.
However if you have a strong relationship to the `ProjectDebugFile`, or an `ArtifactBundle`, that junk is at least being cleaned up once the debug file expires and is being cleaned up.